File: android/Reguerta/app/src/main/java/com/reguerta/user/data/shiftplanning/ChainedShiftPlanningRequestRepository.kt
package com.reguerta.user.data.shiftplanning

import com.reguerta.user.domain.shifts.ShiftPlanningRequest
import com.reguerta.user.domain.shifts.ShiftPlanningRequestRepository

class ChainedShiftPlanningRequestRepository(
    private val primary: ShiftPlanningRequestRepository,
    @Suppress("UNUSED_PARAMETER") fallback: ShiftPlanningRequestRepository,
) : ShiftPlanningRequestRepository {
    override suspend fun submitShiftPlanningRequest(request: ShiftPlanningRequest): ShiftPlanningRequest =
        primary.submitShiftPlanningRequest(request)
}

File: ios/Reguerta/Reguerta/Data/ShiftPlanningRequests/ChainedShiftPlanningRequestRepository.swift
import Foundation

struct ChainedShiftPlanningRequestRepository<
    Primary: ShiftPlanningRequestRepository,
    Fallback: ShiftPlanningRequestRepository
>: ShiftPlanningRequestRepository {
    let primary: Primary
    let fallback: Fallback

    func submit(request: ShiftPlanningRequest) async throws -> ShiftPlanningRequest {
        try await primary.submit(request: request)
    }
}
